src/dune_engine/dune_project.ml

```diff
@@ -816,6 +816,23 @@ let parse ~dir ~lang ~opam_packages ~file ~dir_status =
            Dialect.DB.add dialects ~loc dialect)
          ~init:Dialect.DB.builtin dialects
      in
+     let () =
+       match name with
+       | Named _ -> ()
+       | Anonymous _ ->
+         if
+           dune_version >= (2, 8)
+           && generate_opam_files
+           && dir_status = Sub_dirs.Status.Normal
+         then
+           let loc = Loc.in_file (Path.source file) in
+           User_warning.emit ~loc
+             [ Pp.text
+                 "Project name is not specified. Add a (name <project-name>) \
+                  field to your dune-project file to make sure that $ dune \
+                  subst works in production builds"
```

Comment:
```suggestion
                  subst works (for example for release builds or when package in pinned by opam)"
```

Reply:
I reworded to mention release builds and pins without the parens